express-notes-api/index.js

```diff
@@ -0,0 +1,100 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+const jsonMethod = express.json();
+app.use(jsonMethod);
+
+const data = require('./data.json');
+
+app.get('/api/notes', (req, res) => {
+  const dataArray = [];
+  for (const key in data.notes) {
+    if (data.notes[key]) {
+      dataArray.push(data.notes[key]);
+    }
+  }
+  res.status(200).json(dataArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const iD = req.params.id;
+  if (data.notes[iD]) {
+    if ((Math.sign(iD) === 1) && Number.isInteger(parseInt(iD)) && iD !== '0') {
+      res.status(200).send(data.notes[iD]);
+    }
+    res.status(404).send({ error: `cannot find note with id ${iD}` });
+  }
+  res.status(400).send({ error: 'id must be a positive integer' });
+}
+);
+
+app.post('/api/notes', (req, res) => {
+  const postBody = req.body;
+  const iD = data.nextId;
+  if (!postBody.content) {
+    res.status(400).send({ error: 'content is a required field' });
+  } else if (postBody.content) {
+    postBody.id = iD;
+    data.notes[iD] = postBody;
+    data.nextId++;
+    const newDataJSON = JSON.stringify(data, null, 2);
+    fs.writeFile('./data.json', newDataJSON, err => {
+      if (err) {
+        console.error(err);
+        res.status(500).send({ error: 'An unexpected error occured.' });
+      } else {
+        res.status(201).send(postBody);
+      }
+    });
+  }
+});
+
+app.delete('/api/notes/:id', (req, res) => {
+  const iD = req.params.id;
+  if (isNaN(parseInt(iD))) {
+    res.status(400).send({ error: 'id must be a positive integer' });
```

Comment:
Like the `GET` middleware further up, any `id` which isn't a positive integer should trigger a `400 Bad Request` response: https://en.wikipedia.org/wiki/Natural_number